deploy/Deploy.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.0;
+
+import { OptimismReceiver } from 'lib/xchain-helpers/src/receivers/OptimismReceiver.sol';
+
+import { Executor } from 'src/Executor.sol';
+
+library Deploy {
+
+    function deployExecutor(uint256 delay, uint256 gracePeriod)
+        internal returns (address executor)
+    {
+        executor = address(new Executor(delay, gracePeriod));
+    }
+
+    function deployOptimismReceiver(address l1Authority, address executor)
+        internal returns (address receiver)
+    {
+        receiver = address(new OptimismReceiver(l1Authority, executor));
+    }
+
+    function setUpExecutorPermissions(address executor_, address receiver)
+        internal
+    {
+        // NOTE: Using implementation instead of interface because OZ didn't define
+        //       DEFAULT_ADMIN_ROLE in the IAccessControl interface.
+        Executor executor = Executor(executor_);
+
+        executor.grantRole(executor.SUBMISSION_ROLE(),     receiver);
+        executor.revokeRole(executor.DEFAULT_ADMIN_ROLE(), address(this));
```

Comment:
Deployer address should be passed in as argument rather than `address(this)` which is not the deployer key. The argument that should be passed in inside `Deploy.s.sol` should be `msg.sender` to get the deployer key.